src/modules/explorer/reducers.js

```diff
@@ -245,6 +288,86 @@ export const explorer = (
+            return [];
+        }
+
+        return result.biosamples_with_experiments.flatMap((sample) => {
+            const experiment = sample.experiment;
+            if (!experiment || experiment.experiment_id === null) {
+                return [];
+            }
+
+            const formattedResult = {
+                subjectId: result.subject_id,
+                key: experiment.experiment_id,
+                alternateIds: result.alternate_ids,
+                experimentId: experiment.experiment_id,
+                experimentType: experiment.experiment_type,
+                studyType: experiment.study_type,
+                biosampleId: sample.biosample_id,
+                individual: {
+                    id: result.subject_id,
+                    alternate_ids: result.alternate_ids ?? [],
+                },
+            };
+
+            return formattedResult;
+        });
+    });
+};
+
+function generateBiosampleObjects(searchResults) {
+    const data = (searchResults || {}).results || [];
+    return data
+        .flatMap((result) => {
+            if (!result.biosamples_with_experiments) {
+                return [];
+            }
+
+            const biosampleIdToIndex = {};
+
+            return result.biosamples_with_experiments.reduce((objects, biosample) => {
+                const biosampleId = biosample.biosample_id;
+
+                if (biosampleId) {
+                    const index =
+                        biosampleId in biosampleIdToIndex
+                            ? biosampleIdToIndex[biosampleId]
+                            : (biosampleIdToIndex[biosampleId] = objects.length);
+                    objects[index] = objects[index] || {
+                        subjectId: result.subject_id,
+                        key: biosampleId,
+                        biosample: biosampleId,
+                        alternateIds: result.alternate_ids,
+                        individual: {
+                            id: result.subject_id,
+                            alternateIds: result.alternate_ids || [],
+                        },
+                        experimentIds: [],
+                        experimentTypes: [],
+                        studyTypes: [],
+                        sampledTissues: [],
+                    };
+                    objects[index].experimentIds.push(biosample.experiment.experiment_id);
+                    objects[index].experimentTypes.push(biosample.experiment.experiment_type);
+                    objects[index].studyTypes.push(biosample.experiment.study_type);
+                    objects[index].sampledTissues.push(biosample.sampled_tissue);
+                }
+                return objects;
+            }, []);
+        })
+        .filter((entry) => entry.key !== null && entry.key !== undefined);
+}
+
+const tableSearchResultsBiosamples = (searchResults) => {
+    const res = generateBiosampleObjects(searchResults);
+    return res;
```

Comment:
this function does nothing except call `generateBiosampleObjects` which means you don't need `tableSearchResultsBiosamples` at all